src/spec.c

```diff
@@ -1089,15 +1095,21 @@ static void IndexSpec_FreeTask(IndexSpec *spec) {
   RedisModule_Log(NULL, "notice", "Freeing index %s in background", spec->name);
 #endif
 
-  RedisModuleCtx *threadCtx = RedisModule_GetThreadSafeContext(NULL);
-  RedisModule_AutoMemory(threadCtx);
-  RedisModule_ThreadSafeContextLock(threadCtx);
+  RedisModule_ThreadSafeContextLock(RSDummyContext);
+  RedisModule_AutoMemory(RSDummyContext);
```

Comment:
Again I guess its not related to this PR, lets avoid it in the future and make it in its own PR.